Certificage Manager Crash - Assignment to entry in nil map

**Describe the bug**
After upgrading from 2.0.0 straight to the latest development version of skupper I encountered this panic on controller startup:

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/skupperproject/skupper/internal/kube/certificates.(*CertificateManagerImpl).updateSecret(0xc0002ffc00, {0xc000da6a38, 0x17}, 0xc000c46d00, 0xc000015a40)
        /go/src/app/internal/kube/certificates/mgr.go:296 +0x19d
github.com/skupperproject/skupper/internal/kube/certificates.(*CertificateManagerImpl).reconcile(0xc0002ffc00, {0xc000da6a38?, 0xc000da6a38?}, 0xc000c46d00, 0xc0000baa90?)
        /go/src/app/internal/kube/certificates/mgr.go:238 +0x2b
github.com/skupperproject/skupper/internal/kube/certificates.(*CertificateManagerImpl).checkCertificate(0xc0002ffc00, {0xc000da6a38, 0x17}, 0xc000c46d00)
        /go/src/app/internal/kube/certificates/mgr.go:228 +0x69
github.com/skupperproject/skupper/internal/kube/certificates.(*CertificateManagerImpl).Recover(0xc0002ffc00)
        /go/src/app/internal/kube/certificates/mgr.go:96 +0x2a7
github.com/skupperproject/skupper/internal/kube/controller.(*Controller).init(0xc0001583c0, 0xc0004de0e0)
        /go/src/app/internal/kube/controller/controller.go:301 +0x1465
github.com/skupperproject/skupper/internal/kube/controller.(*Controller).Run(0xc0001583c0, 0xc0004de0e0)
        /go/src/app/internal/kube/controller/controller.go:192 +0x1d
main.main()
        /go/src/app/cmd/controller/main.go:76 +0x518

```

**How To Reproduce**
I am not sure how we got here: There exists a a "skupper-site-server" Secret with its owner set to the corresponding Certificate but with no annotations set. When the certificate manager attempts to recover this certificate it panics.

```
$ k get secrets -n foo skupper-site-server -oyaml | yq .metadata.ownerReferences
- apiVersion: skupper.io/v2alpha1
  kind: Certificate
  name: skupper-site-server
  uid: 6bc46b50-1e32-4670-a1bd-b03734a8ddc5
$ k get secrets -n foo skupper-site-server -oyaml | yq .metadata.annotations
null
```

**Environment details**
Skupper 2.0.0 to latest development

**Additional context**
Add any other context about the problem here.

--- a/internal/kube/certificates/mgr.go
+++ b/internal/kube/certificates/mgr.go
@@ -293,6 +293,9 @@ func (m *CertificateManagerImpl) updateSecret(key string, certificate *skupperv2
 		}
 		changed = true
 		secret.Data = regenerated.Data
+		if secret.Annotations == nil {
+			secret.Annotations = map[string]string{}
+		}
 		secret.Annotations["internal.skupper.io/hosts"] = strings.Join(certificate.Spec.Hosts, ",")
 	}
 	if m.context != nil && controlled {

--- a/internal/kube/certificates/mgr_test.go
+++ b/internal/kube/certificates/mgr_test.go
@@ -243,6 +243,23 @@ func TestCertificateManager(t *testing.T) {
 			expectedSecrets:      []*corev1.Secret{},
 			expectedCertificates: []*skupperv2alpha1.Certificate{},
 		},
+		{
+			name: "owned secret missing annotations",
+			k8sObjects: []runtime.Object{
+				myCaFixture,
+				withOwnerRef(secret("foo", "test", nil, nil, nil), metav1.OwnerReference{APIVersion: "skupper.io/v2alpha1", Kind: "Certificate", Name: "foo"}),
+			},
+			skupperObjects: []runtime.Object{
+				certificate("foo", "test", "my-ca", "my-subject", []string{"aaa", "bbb"}, false, true, nil, nil),
+			},
+			expectedSecrets: []*corev1.Secret{
+				secret("foo", "test", nil, nil, nil),
+				secret("my-ca", "test", nil, nil, nil),
+			},
+			expectedCertificates: []*skupperv2alpha1.Certificate{
+				addCertificateStatus(certificate("foo", "test", "my-ca", "my-subject", []string{"aaa", "bbb"}, false, true, nil, nil), "", "", condition(skupperv2alpha1.CONDITION_TYPE_READY, metav1.ConditionTrue, "Ready", "OK")),
+			},
+		},
 	}
 	for _, tt := range testTable {
 		t.Run(tt.name, func(t *testing.T) {
@@ -322,6 +339,10 @@ func secret(name string, namespace string, data map[string][]byte, labels map[st
 	}
 }
 
+func withOwnerRef(secret *corev1.Secret, ref metav1.OwnerReference) *corev1.Secret {
+	secret.ObjectMeta.OwnerReferences = []metav1.OwnerReference{ref}
+	return secret
+}
 func certificate(name string, namespace string, ca string, subject string, hosts []string, client bool, server bool, labels map[string]string, annotations map[string]string) *skupperv2alpha1.Certificate {
 	return &skupperv2alpha1.Certificate{
 		TypeMeta: metav1.TypeMeta{

